src/main/java/dev/blaauwendraad/masker/json/KeyContainsMasker.java

```diff
@@ -164,38 +161,103 @@ public byte[] mask(byte[] input) {
             }
             // else, if nothing has matched, continue to the next value
             // it may happen that we don't mask anything if number / boolean masking is disabled or the value is null
-
-            maskingState.backtrackCurrentJsonPath();
         }
 
         ValueMaskingUtil.flushReplacementOperations(maskingState);
 
         return maskingState.getMessage();
     }
 
+    /**
+     * Masks the array element in place.
+     *
+     * @param maskingState the masking state an index prior to the element start index
+     */
+    private void maskArrayElementInPlace(MaskingState maskingState, KeyMaskingConfig keyMaskingConfig) {
+        maskingState.incrementCurrentIndex();
+        skipWhitespaceCharacters(maskingState);
+        boolean maskable = maskValue(maskingState, keyMaskingConfig);
+        if (!maskable) {
+            // non-maskable values
+            skipAllValues(maskingState);
+        }
+    }
+
+    /**
+     * Masks the value in a key/value pair where the current index is on the value start
+     *
+     * @param maskingState     the current masking state in which the value will be masked
+     * @param keyMaskingConfig the masking configuration for the key
+     * @return true if the value is maskable, false otherwise
+     */
+    private boolean maskValue(MaskingState maskingState, KeyMaskingConfig keyMaskingConfig) {
+        if (AsciiCharacter.isSquareBracketOpen(maskingState.byteAtCurrentIndex())) {
+            maskArrayValueInPlace(maskingState, keyMaskingConfig);
+        } else if (AsciiCharacter.isDoubleQuote(maskingState.byteAtCurrentIndex())) {
+            maskStringValueInPlace(maskingState, keyMaskingConfig);
+        } else if (AsciiJsonUtil.isFirstNumberChar(maskingState.byteAtCurrentIndex())
+                && !keyMaskingConfig.isDisableNumberMasking()) {
+            maskNumberValueInPlace(maskingState, keyMaskingConfig);
+        } else if (AsciiCharacter.isCurlyBracketOpen(maskingState.byteAtCurrentIndex())) {
+            maskObjectValueInPlace(maskingState, keyMaskingConfig);
+        } else if ((AsciiCharacter.isLowercaseF(maskingState.byteAtCurrentIndex())
+                || AsciiCharacter.isLowercaseT(maskingState.byteAtCurrentIndex()))
+                && !keyMaskingConfig.isDisableBooleanMasking()) {
+            maskBooleanValueInPlace(maskingState, keyMaskingConfig);
+        } else {
+            return false;
+        }
+        return true;
+    }
+
     /**
      * Checks if the current byte changes the current json path. If so, updates current json path.
```

Comment:
the current byte => 'the current byte in the maskingState` 

Also "changes" is not super clear from JavaDoc what this means

Reply:
I will fix it.